dispatcher/pathbased.go

```diff
@@ -127,6 +128,17 @@ func (d *pathBasedDispatcher) buildOptionsHandler(cfg *config.Application, inner
 		}
 
 		rw.Header().Set("Allow", allow)
+
+		body, err := ioutil.ReadAll(recorder.Body)
+		if err != nil {
+			d.log.Errorf("error while reading response body: %s", err)
+			rw.WriteHeader(500)
+			contentLength, _ := rw.Write([]byte(`{"msg":"internal server error"}`))
+			rw.Header().Set("Content-Length", fmt.Sprintf("%d", contentLength))
+			return
+		}
+		rw.Write(body)
```

Comment:
Maybe easier (untested, though)?

```
length, err := io.Copy(recorder.Body, rw)
```

Reply:
Looks good.
The first param is destination, second is the source so is switched the params.

Reply:
Ouch, good catch. :+1: